src/work_pool.c

```diff
@@ -206,25 +209,23 @@ work_pool_thread(void *arg)
 		 */
 		rc = pthread_cond_timedwait(&wpt->pqcond, &pool->pqh.qmutex,
 					    &ts);
-		if (rc) {
```

Comment:
grr github, I can't comment on lines out of context -- a few lines above this is `imespec_addms(&ts, WORK_POOL_TIMEOUT_MS)`, that needs to use `pool->timeout`

Reply:
ugh not sure why github still displays that, code looks fine once pulled in....

Reply:
Sorry, I tested incrementally, didn't realize you were reviewing incrementally. 
I have a tendency to push over and over....

At this point, compiled without ASAN, and pynfs runs followed by clean exit.

Reply:
No problem, I think I just need to close the github window and reload when I take a fresh look as the dynamic stuff doesn't work well -- some of the code I was seeing definitely was new code, it's weird. Anyway thanks for looking at it!